src/records/config/index.test.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  mapVariants,
+  mergeTestConfig,
+  mergeDefaultConfig,
+  defaultTestConfig,
+  defaultConfig,
+} from ".";
+
+const STANDARD_VARIANTS = {
+  VALUE_A: {
+    value: "value_A",
+    ratio: 2,
+  },
+  VALUE_B: {
+    value: "value_B",
+    ratio: 5,
+  },
+};
+
+const NUMBER_VARIANTS = {
+  VALUE_C: 3,
+  VALUE_D: 10,
+};
+
+const NUMBER_VARIANTS_STANDARDISED = {
+  VALUE_C: {
+    value: "VALUE_C",
+    ratio: 3,
+  },
+  VALUE_D: {
+    value: "VALUE_D",
+    ratio: 10,
+  },
+};
+
+const VARIANTS = { ...NUMBER_VARIANTS, ...STANDARD_VARIANTS };
+const VARIANTS_STANDARDISED = {
+  ...NUMBER_VARIANTS_STANDARDISED,
+  ...STANDARD_VARIANTS,
+};
+
+describe("mapVariants", () => {
+  test("should convert variants to a standard form", () => {
+    expect(mapVariants(STANDARD_VARIANTS)).toStrictEqual(STANDARD_VARIANTS);
+    expect(mapVariants(NUMBER_VARIANTS)).toStrictEqual(NUMBER_VARIANTS_STANDARDISED);
+    expect(mapVariants(VARIANTS)).toStrictEqual(VARIANTS_STANDARDISED);
+  });
+  test("should return an empty object for unexpected inputs", () => {
```

Comment:
not sure if I like that it does some magic instead of throwing error, but that's for another  discussion

Reply:
Yes, there's a lot of magic that I'd like to get rid of in future PRs.